### ISSUE TITLE
After upgrade to 3.x, the Instrumentation of .net core 3.1 background service (job) fails with ArgumentException

### ISSUE THREAD

**Issue by lpodolak** [DataDog/dd-trace-dotnet#6073]
**Describe the bug**

We have couple of services (most of these are web apis) and jobs. All run in Kubernetes (GKE). Unfortunately, due to lack of permissions for GKE nodes, all our apps are auto-instrumented from the datadog-agent sidecar container. Recently I upgraded both: Tracer and Datadog.Trace from 2.57.0 (all was working fine) to versions of 3.x (I tried 3.2.0 as well as the latest atm: 3.3.1). Our web APIs work without any problems after that upgrade, however jobs (running As CronJobs in GKE) started to crash right at the start. I am not entirely sure what happens, but it looks like the exception comes from IL-generaged code crashes entire job app. I even don't know how to catch that exception, so I can't use datadog anymore with that version. 

Below is the error that we get:
```
System.ArgumentException: Expression of type 'Datadog_Trace__DEF86D061D0D2EEB.Datadog_Trace_Scope.Datadog_Trace_IScope_26' cannot be used for parameter of type 'Datadog.Trace.IScope' of method 'Void set_DatadogScope(Datadog.Trace.IScope)' (Parameter 'arg0')
   at System.Dynamic.Utils.ExpressionUtils.ValidateOneArgument(MethodBase method, ExpressionType nodeKind, Expression arguments, ParameterInfo pi, String methodParamName, String argumentParamName, Int32 index) in /_/src/System.Linq.Expressions/src/System/Dynamic/Utils/ExpressionUtils.cs:line 184
   at System.Linq.Expressions.Expression.Call(Expression instance, MethodInfo method, Expression arg0) in /_/src/System.Linq.Expressions/src/System/Linq/Expressions/MethodCallExpression.cs:line 1067
   at System.Linq.Expressions.Expression.Call(Expression instance, MethodInfo method, IEnumerable`1 arguments) in /_/src/System.Linq.Expressions/src/System/Linq/Expressions/MethodCallExpression.cs:line 1204
   at Lamar.IoC.Instances.InstanceConstructorFrame.WriteExpressions(LambdaDefinition definition)
   at Lamar.IoC.Instances.FuncResolverDefinition.BuildResolver()
   at Lamar.IoC.Instances.GeneratedInstance.BuildFuncResolver(Scope scope)
   at Lamar.IoC.Instances.GeneratedInstance.buildResolver(Scope scope)
   at Lamar.IoC.Instances.GeneratedInstance.ToResolver(Scope topScope)
   at Lamar.IoC.Instances.ConstructorInstance.ToResolver(Scope topScope)
   at Lamar.ServiceGraph.FindResolver(Type serviceType)
   at Lamar.IoC.Scope.GetInstance(Type serviceType)
   at Lamar.IoC.Scope.GetInstance[T]()
   at xxx.Jobs.JobExecutor.ExecuteAsync[TJobArgs,TGlobalSettings](TJobArgs jobArgs, IContainer container, IJobGlobalSettings gs, Action`1 publishDatadogRootTrace)
   at xxx.Jobs.Program.Main(String[] args) in /app/xxx.Jobs/Program.cs:line 55
```

**To Reproduce**
1. Create .net core 3.1 console app, create a Job in GKE. Set the datadog agent in sidecar container that auto-instruments the actual job container
3. There is no OOTB (or at least I know) way to achieve instrumentation of non-http service (like a job), so Let's create a root span![image](https://github.com/user-attachments/assets/1b044bc5-40a6-4515-91a8-c0d2bf739654)
6. Run the Job
7. Observe the job crashes at start producing the mentioned error.

**Expected behavior**
Custom instrumentation sends spans correctly so that they are visible in Datadog UI and don't crash entire app, as in version 2.x (verified with 2.57.0 and couple of earlier versions as well)

**Screenshots**
Container structure in Pod
![image](https://github.com/user-attachments/assets/f8dba237-cb38-4725-8f4c-ef2b373ffe57)
Datadog Agent configuration (in sidecar pattern)
![image](https://github.com/user-attachments/assets/bf1c98da-5c93-47dd-b2a6-6c7688085519)
The Job container configuration 
![image](https://github.com/user-attachments/assets/f8aed781-4c05-4b82-a401-a842e328eb25)

**Runtime environment (please complete the following information):**
- Instrumentation mode: automatic via datadog-agent (via sidecar container) and custom (with Datadog.Trace nuget)
- Tracer version: 3.x (verified failures in 3.2.0 and 3.3.1)
- OS: Dockerized: Red Hat Enterprise Linux 8.10
- CLR: .net core 3.1

**Additional context**
So far what I recognized is that the combination of the Tracer software (installed on the docker image) in version 3.x works as long as the Datadog.Trace nuget package remains in 2.57.0, that's why my first feeling is that the problem is in theDatadog.Trace nuget package on the .net core 3.1 platform. Unfortunately, I can't test this code on newer .net core versions - we plan to do the migration later this year, but now it is impossible to upgrade. From what I read in release changelog, you guys insist on having these both components in same versions, which makes me even more confused that such combination works. 

<!--
NOTE: To contact our support team, see https://www.datadoghq.com/support/
When contacting support, please include the information requested above.
If possible, also include the tracer log files found (by default) in:
Windows: %ProgramData%\Datadog .NET Tracer\logs\
Linux: /var/log/datadog/dotnet/
--!>

**Comment by andrewlock** [comment]
Hi @lpodolak, thanks for flagging this. The issue appears to be a curious consequence of us making the return type from `Tracer.Instance.StartActive()` a `struct` instead of `class` in v3. Even though the returned type _does_ implement `IScope`, there's [a curious check](https://github.com/dotnet/runtime/blob/cf258a14b70ad9069470a108f13765e0e5988f51/src/libraries/System.Linq.Expressions/src/System/Dynamic/Utils/TypeUtils.cs#L851) in the expression building code that causes it to fail for structs 🤷‍♂️ 

https://github.com/DataDog/dd-trace-dotnet/pull/6082 should resolve the issue, which should be included in the next release. Thanks again!

**Comment by andrewlock** [comment]
Hi @lpodolak this should be fixed in v3.4.0. Thanks!

**Comment by lpodolak** [comment]
Thank you very much, it works now with v3.4.0
